### PR TITLE
Mixins: check that `memcached_index_writes` exists before using it in `boltdb_shipper`

### PR DESCRIPTION
**Pull request by dannykopping**
**What this PR does / why we need it**:
This key is missing when not using boltdb_shipper, but this lib is included by default

### CODE REVIEW

File: CHANGELOG.md [hunk 1]
```diff
@@ -24,6 +24,14 @@
 
 * [8988](https://github.com/grafana/loki/pull/8988) **darxriggs**: Promtail: Prevent logging errors on normal shutdown.
 
+#### Mixins
+
+#### Enhancements
+
+#### Fixes
+
+* [8995](https://github.com/grafana/loki/pull/8995) **dannykopping**: Mixins: check that `memcached_index_writes` exists before using it in `boltdb_shipper`
```
Comment (chaudum) [inline CHANGELOG.md:33]: What user facing error/bug does this change fix?

Comment (dannykopping) [reply CHANGELOG.md:33]: You will get

```
evaluating jsonnet: RUNTIME ERROR: Field does not exist: memcached_index_writes
```

if you try and use the mixins with `use_boltdb_shipper: false`.

Comment (chaudum) [reply CHANGELOG.md:33]: I'd phrase the changelog entry accordingly then:

```suggestion
* [8995](https://github.com/grafana/loki/pull/8995) **dannykopping**: Mixins: Fix Jsonnet `RUNTIME ERROR` that occurs when you try to use the mixins with `use_boltdb_shipper: false`.
```

Comment (dannykopping) [reply CHANGELOG.md:33]: Good call, thanks!

**Review by chaudum — APPROVED** [review]

**Pull request merged**